Content.Server/Research/Systems/RoboticsConsoleSystem.cs

```diff
@@ -0,0 +1,157 @@
+        {
+            subs.Event<BoundUIOpenedEvent>(OnOpened);
+            subs.Event<RoboticsConsoleDisableMessage>(OnDisable);
+            subs.Event<RoboticsConsoleDestroyMessage>(OnDestroy);
+            // TODO: camera stuff
+        });
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var now = _timing.CurTime;
+        foreach (var comp in EntityQuery<RoboticsConsoleComponent>())
+        {
+            // remove cyborgs that havent pinged in a while
+            _removing.Clear();
+            foreach (var (address, timeout) in comp.Timeouts)
+            {
+                if (now >= timeout)
+                    _removing.Add(address);
+            }
+
+            // needed to prevent modifying while iterating it
+            foreach (var address in _removing)
+            {
+                comp.Cyborgs.Remove(address);
+                comp.Timeouts.Remove(address);
+            }
+        }
+    }
+
+    private void OnUnpaused(Entity<RoboticsConsoleComponent> ent, ref EntityUnpausedEvent args)
+    {
+        ent.Comp.NextDestroy += args.PausedTime;
+    }
+
+    private void OnPacketReceived(Entity<RoboticsConsoleComponent> ent, ref DeviceNetworkPacketEvent args)
+    {
+        var payload = args.Data;
+        if (!payload.TryGetValue(DeviceNetworkConstants.Command, out string? command))
+            return;
+        if (command != DeviceNetworkConstants.CmdUpdatedState)
+            return;
+
+        if (!payload.TryGetValue(RoboticsConsoleConstants.NET_CYBORG_DATA, out CyborgControlData? data))
+            return;
+
+        var address = args.SenderAddress;
+        ent.Comp.Cyborgs[address] = data.Value;
+        ent.Comp.Timeouts[address] = _timing.CurTime + ent.Comp.Timeout;
+
+        UpdateUserInterface(ent);
+    }
+
+    private void OnOpened(Entity<RoboticsConsoleComponent> ent, ref BoundUIOpenedEvent args)
+    {
+        UpdateUserInterface(ent);
+    }
+
+    private void OnDisable(Entity<RoboticsConsoleComponent> ent, ref RoboticsConsoleDisableMessage args)
+    {
+        if (args.Session.AttachedEntity is not {} user || !_access.IsAllowed(user, ent))
+            return;
+
+        if (!ent.Comp.Cyborgs.TryGetValue(args.Address, out var data))
+            return;
+
+        var payload = new NetworkPayload()
+        {
+            [DeviceNetworkConstants.Command] = RoboticsConsoleConstants.NET_DISABLE_COMMAND
+        };
+
+        _deviceNetwork.QueuePacket(ent, args.Address, payload);
+        _adminLogger.Add(LogType.RoboticsConsole, LogImpact.High, $"{ToPrettyString(user):user} disabled borg {data.Name} with address {args.Address}");
+    }
+
+    private void OnDestroy(Entity<RoboticsConsoleComponent> ent, ref RoboticsConsoleDestroyMessage args)
+    {
+        if (args.Session.AttachedEntity is not {} user || !_access.IsAllowed(user, ent))
+            return;
+
```

Comment:
See https://github.com/space-wizards/space-station-14/issues/24803

Reply:
this would be fine for locking the console itself but rd accidentally leaving it unlocked means shitters can start destroying borgs which is bad